mmdet/models/roi_heads/sparse_roi_head.py

```diff
@@ -259,8 +301,13 @@ def simple_test(self,
         # Decode initial proposals
         num_imgs = len(img_metas)
         proposal_list = [proposal_boxes[i] for i in range(num_imgs)]
-        object_feats = proposal_features
+        ori_shapes = tuple(meta['ori_shape'] for meta in img_metas)
+        scale_factors = tuple(meta['scale_factor'] for meta in img_metas)
+        # "ms" in variable names means multi-stage
+        ms_bbox_result = {}
+        ms_segm_result = {}
 
+        object_feats = proposal_features
         if all([proposal.shape[0] == 0 for proposal in proposal_list]):
```

Comment:
In the subsequent code, you still need to deal with the empty proposal scene in a single image.

Reply:
Firstly, query plays the same role in QueryInst as anchors in FasterR-CNN, so if there is no queries in QueryInst or SparseRCNN, just like no anchor bboxes in FasterR-CNN, this kind of detectors are meaningless. Secondly, for mask head, if we use random crop data augmentation, the empty proposal may come to the mask forward process, and, actually we have already considered this exception. Finally, if you really want to deal with this exceptional case, I prefer to use assert to make sure it will never happen. 
Glad to discuss with you and hope for your reply.

Reply:
From an overall point of view, this situation does not exist, but if users freely combine different algorithm modules to call this situation, I hope that no error will be reported. If this hidden danger has been discovered, my suggestion is to avoid it. If you don't think it makes sense, then assert is also an optional solution. In short, I don’t want to report errors directly in any scenario. This is just my personal opinion and glad to discuss with you.